sct-commons/src/test/java/org/lfenergy/compas/sct/commons/ExtRefEditorServiceTest.java

```diff
@@ -955,4 +956,44 @@ void updateIedNameBasedOnLnode_when_no_Compas_ICD_Header_should_return_an_error(
                 .containsExactly(Tuple.tuple(true,
                         "The substation LNode with following attributes : IedName:IED_NAME2 / LdInst:LD_INST21 / LnClass:ANCR / LnInst:1  does not contain the needed (COMPAS - ICDHeader) private"));
     }
+
+    @Test
+    void epfPostProcessing_when_exist_unused_channel_should_update_setSrcRef() {
+        // Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scd-ldepf/scd_ldepf_postProcessing.xml");
+        // When
+        extRefEditorService.epfPostProcessing(scd);
+        // Then
+        SoftAssertions softly = new SoftAssertions();
+
+        Optional<TDAI> setSrcRefInInRef1 = findDai(scd, "IED_NAME1", "LDEPF", "InRef1", "setSrcRef");
+        Optional<TDAI> purposeInInRef1 = findDai(scd, "IED_NAME1", "LDEPF", "InRef1", "purpose");
+        assertThat(purposeInInRef1).isPresent();
+        softly.assertThat(purposeInInRef1.get().getVal().getFirst().getValue()).doesNotStartWith("DYN_LDEPF_DIGITAL CHANNEL");
+        softly.assertThat(purposeInInRef1.get().getVal().getFirst().getValue()).doesNotStartWith("DYN_LDEPF_ANALOG CHANNEL");
+        assertThat(setSrcRefInInRef1).isPresent();
+        softly.assertThat(setSrcRefInInRef1.get().isSetVal()).isFalse();
+
+        Optional<TDAI> setSrcRefInInRef2 = findDai(scd, "IED_NAME1", "LDEPF", "InRef2", "setSrcRef");
+        Optional<TDAI> purposeInInRef2 = findDai(scd, "IED_NAME1", "LDEPF", "InRef2", "purpose");
+        assertThat(purposeInInRef2).isPresent();
+        softly.assertThat(purposeInInRef2.get().getVal().getFirst().getValue()).startsWith("DYN_LDEPF_DIGITAL CHANNEL");
+        assertThat(setSrcRefInInRef2).isPresent();
+        softly.assertThat(setSrcRefInInRef2.get().getVal().getFirst().getValue()).isEqualTo("IED_NAME1LDEPF/LPHD0.Proxy");
+
+        Optional<TDAI> setSrcRefInInRef3 = findDai(scd, "IED_NAME1", "LDEPF", "InRef3", "setSrcRef");
+        Optional<TDAI> purposeInInRef3 = findDai(scd, "IED_NAME1", "LDEPF", "InRef3", "purpose");
+        assertThat(purposeInInRef3).isPresent();
+        softly.assertThat(purposeInInRef3.get().getVal().getFirst().getValue()).startsWith("DYN_LDEPF_DIGITAL CHANNEL");
```

Comment:
could we use equals instead of stratsWith to differenciate the test from the one above easily ?

Reply:
It's not possible, there are two objects in different InRef having value that start with the same text.
The only difference is that the first object's Val section is missing, while the second is empty.